mantidproject/mantid
# RROA Polref interface: Add global options to settings tab

As described in the issue, this PR adds several settings to the settings tab. In the original interface, these settings were all specified together for each algorithm as keyword arguments. Now, each setting is given their own edit space and are split between 'Experiment' and 'Instrument' sections. In addition, a 
'Get Defaults' button has been added for each section that fills the settings with default values. For experiment settings, default values are pulled from the RROA algorithm whereas for instrument settings, they are pulled from the current instrument being used in the run.

The list of settings added are as follows:

Experiment settings:
- AnalysisMode
- DirectBeam
- Transmission run(s)
- StartOverlap
- EndOverlap
- PolarisationCorrections
- CRho
- CAlpha
- CAp
- CPp
- dQ/Q
- Scale
- Stitch1DMany

Instrument settings:
- IntegratedMonitors
- MonitorIntegralMin
- MonitorIntegralMax
- MonitorBackgroundMin
- MonitorBackgroundMax
- LambdaMin
- LambdaMax
- I0MonitorIndex
- ProcessingInstructions

Original interface (Global options) has been left in for now.

**To test:**

Ensure all unit tests pass.

Ensure that altering the settings from both experiment and instrument settings affect the algorithm runs correctly.

Ensure the 'Get Defaults' buttons for each section works as described. For `Instrument Settings' section, ensure it obtains the correct default values for each available instrument (set in the runs tab).

Fixes #16983.

Release notes have been updated
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
